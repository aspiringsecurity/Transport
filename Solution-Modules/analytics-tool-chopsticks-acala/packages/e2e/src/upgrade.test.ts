import { afterAll, describe, expect, it } from 'vitest'
import { readFileSync } from 'node:fs'
import path from 'node:path'

import { expectJson, testingPairs } from './helper'

import networks from './networks'

describe('upgrade', async () => {
  const { alice, bob } = testingPairs()
  const acala = await networks.acala({
    blockNumber: 2000000,
  })
  const { api, dev, chain } = acala

  afterAll(async () => {
    await acala.teardown()
  })

  it('setCode works', async () => {
    await dev.setStorage({
      Sudo: {
        Key: alice.address,
      },
      System: {
        Account: [[[alice.address], { data: { free: 1000 * 1e12 } }]],
      },
    })

    const runtime = readFileSync(path.join(__dirname, '../blobs/acala-runtime-2101.txt')).toString().trim()

    expect(await chain.head.runtimeVersion).toContain({ specVersion: 2096 })
    await api.tx.sudo.sudoUncheckedWeight(api.tx.system.setCode(runtime), '0').signAndSend(alice)
    await dev.newBlock({ count: 3 })
    expect(await chain.head.runtimeVersion).toContain({ specVersion: 2101 })
    expect(api.runtimeVersion.specVersion).toMatchInlineSnapshot(`2101`)

    await api.tx.balances.transfer(bob.address, 1e12).signAndSend(alice)
    await dev.newBlock()
    expectJson(await api.query.system.account(bob.address)).toMatchSnapshot()

    await api.tx.balances.transfer(bob.address, 1e12).signAndSend(alice)
    await dev.newBlock()
    expectJson(await api.query.system.account(bob.address)).toMatchSnapshot()
  })
})
