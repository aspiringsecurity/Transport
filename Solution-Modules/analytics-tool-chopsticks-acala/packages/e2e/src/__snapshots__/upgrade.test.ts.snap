// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`upgrade > setCode works 2`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 1000000000000,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`upgrade > setCode works 3`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 2000000000000,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;
